macros/src/main/scala/pureconfig/Derivation.scala

```diff
@@ -50,9 +50,15 @@ class DerivationMacros(val c: whitebox.Context) extends LazyContextParser with M
 
     // check the `-Xmacro-settings:materialize-derivations` scalac flag
     if (!isDerivationEnabled || !isHeadImplicitADerivation) {
-      // when not present, simply render the base `Derivation` constructor with `implcitly[A]`. This results in the same
-      // behavior as without `Derivation` (apart from the extra wrapper).
-      q"_root_.pureconfig.Derivation.Successful(_root_.scala.Predef.implicitly[${weakTypeOf[A]}])"
+      // when not present, start an implicit search for `A` and place it inside a `Derivation.Successful` if the search
+      // succeeds.
+      val tpe = weakTypeOf[A]
+      inferImplicitValueCompat(tpe) match {
+        case EmptyTree =>
+          c.abort(c.enclosingPosition, s"could not derive ${prettyPrintType(tpe)}")
```

Comment:
As we discussed in private, I find it strange that `materializeDerivation` yields a custom, but not "derivation-like" error when called explicitly.

In the past, we decided to make `materializeDerivation` work just like calling `implicitly` when called explicitly, leading to the strange error we're having. I think that the most expectable behavior is to make `materializeDerivation` do its custom logic when called explicitly, regardless of whether the compiler flag is enabled and yielding always the hierarchical "implicit not found" message.

Reply:
I agree! I think I got `materializeDerivation` to yield the proper derivation-like message in fe5fa43 and 977e0db.